code/modules/holodeck/computer.dm

```diff
@@ -25,44 +31,54 @@
 	idle_power_usage = 10
 	active_power_usage = 50
 
-	var/area/holodeck/linked
-	var/area/holodeck/program
-	var/area/holodeck/last_program
-	var/area/offline_program = /area/holodeck/rec_center/offline
+	var/area/holodeck/linked //the area that this holodeck loads templates into, used for power and deleting holo objects that leave it
+
+	var/area/mappedstartarea = /area/holodeck/rec_center //change this to a different area if youre making a second holodeck different from the station one
+	var/program = "holodeck_offline" //what program is loaded right now
+	var/last_program
 
-	var/list/program_cache
-	var/list/emag_programs
+	var/offline_program = "holodeck_offline" //the default program loaded by this holodeck when spawned and when deactivated
+	var/holodeck_access = STATION_HOLODECK //what access type this holodeck has, used to specify programs for another holodeck that others cant load
 
-	// Splitting this up allows two holodecks of the same size
-	// to use the same source patterns.  Y'know, if you want to.
-	var/holodeck_type = /area/holodeck/rec_center	// locate(this) to get the target holodeck
-	var/program_type = /area/holodeck/rec_center	// subtypes of this (but not this itself) are loadable programs
+	var/list/program_cache //this stores all of the unrestricted holodeck map templates that this computer has access to
+	var/list/emag_programs //as above, but for restricted ones that requires the safeties to be disabled to load
+
+	var/program_type = /datum/map_template/holodeck	// subtypes of this (but not this itself) are loadable programs
+
+	var/list/spawned = list() //every holo object created by the holodeck goes in here to track it
+	var/list/effects = list() //like above, but for holo effects
```

Comment:
Are these things being cleared from the list when destroyed, to avoid hard deletions?

Reply:
when they are deleted by the holodeck manually yes (they leave the holodeck area or the simulation changes, deleting everything from the previous program), otherwise no theyre not cleared when destroyed.

Reply:
Register to their destruction, then, and handle it.

Reply:
done now